### ISSUE TITLE
Pressure Chamber problems with recipe input and output Items that have NBT (added with CraftTweaker)

### ISSUE THREAD

**Issue by Robijnvogel** [TeamPneumatic/pnc-repressurized#1113]
### Describe the bug

Hey there,

I am experiencing this issue on MC 1.12.2, which I realize is a rather old version, which may no longer be supported.
BUT, since I can not find an issue (open or closed) about this in your issue tracker, it may still be present in the currently supported versions of this mod.

If this does no longer occur in currently supported versions of this mod, this issue can at least serve as documentation of that fact. ;)

The issue(s):
I am playing the Project Ozone modpack, to which a [Pressure Chamber recipe](https://imgur.com/uIEiEyp) was added to create a (Thermal Expansion) Florb of Antimatter (Abyssalcraft) recipe by combining 2 simple items and 2 other Florbs with another Abyssal and an Erebus liquid in them.
These Florbs are the Items that I expect make the recipe misbehave in the following ways:
1. It does not matter which Florbs are used for the input. Any Florb, for instance one filled with Water and the 2 simple Items will result in 1 Florb of Antimatter.
2. It does not matter how many Florbs (or other input items are provided. All the items that match the input are consumed, most of the time.
3. The quantity of the output is typically lower than expected.

Some tests I did (once each):
- 1 of everything (technically a Florb short) yielded 1 Florb of Antimatter. Expected: 0 (no crafting)
- 4 Florbs and 4 of each of the input Items yielded 1 Florb of Antimatter. Expected: 2
- 8 Florbs and 5 of each of the input Items yielded 3 Florbs of Antimatter. Expected: 4 (and 1 leftover of each simple item)
- 10 Florbs and 5 of each of the input Items yielded 5 Florbs of Antimatter. Expected: 5 [v]
- 128 Florbs and 64 of each of the input Items yielded 32 Florbs of Antimatter and left 32 Florbs of Water untouched somehow? Expected: 64 and nothing leftover

So weird things are happening. It seems like in some cases the non-NBT Items are being consumed too early/too much resulting in less output Florbs and leftover input Florbs.

Note: Items in the Pressure Chamber that do not match the recipe at all are not consumed. No worries. :)

### How to reproduce the bug

Project Ozone 3, Build a Pressure Chamber with a Creative Compressor and try the recipe for a Florb of Antimatter using Pressure Chamber Interfaces.

Perhaps this can be reproduced in a newer MC version using CraftTweaker and a similar script:
```
#packmode normal titan kappa mythic
mods.pneumaticcraft.pressurechamber.addRecipe([<thermalexpansion:florb>.withTag({Fluid: "liquidcoralium"}), <thermalexpansion:florb>.withTag({Fluid: "formic_acid"}), <lordcraft:crystal_basic_cosmic:0>, <randomthings:ingredient:1>], 2.0, [<thermalexpansion:florb>.withTag({Fluid: "liquidantimatter"})]);
```
I don't know if Thermal Expansion Florbs still use NBT, but since they are added for each fluid dynamically they still may.

### Expected behavior

Input NBT should be evaluated.
Output quantity should be correct.
Remaining Items should not be consumed.

### Which Minecraft version are you using?

1.12 - UNSUPPORTED

### Which version of PneumaticCraft: Repressurized are you using?

1.12.2-0.11.15-398

**Comment by desht** [comment]
Yeah, sorry, I don't touch 1.12.2 at all anymore.

**Comment by Robijnvogel** [comment]
@desht I'm sorry for not being clear about this, but at no point did I expect this to get fixed in 1.12.2.

Does this no longer happen in 1.19?

**Comment by desht** [comment]
I'll take a look at 1.19 and see how it's behaving. There's a good chance it won't happen since there was a major rewrite post 1.12..
